lib/client/api.go

```diff
@@ -796,8 +796,43 @@ func WithMakeCurrentProfile(makeCurrentProfile bool) RetryWithReloginOption {
 	}
 }
 
+// NewNonRetryableError wraps an error to indicate that the error should fail
+// IsErrorResolvableWithRelogin. This wrapper is used to workaround the false
+// positives like trace.IsBadParameter check in IsErrorResolvableWithRelogin.
+func NewNonRetryableError(err error) error {
+	return &nonRetryableError{
+		Err: err,
+	}
+}
+
+type nonRetryableError struct {
+	Err error
+}
+
+func (e *nonRetryableError) Error() string {
+	if e == nil || e.Err == nil {
+		return "error"
+	}
+	return e.Err.Error()
+}
+
+func (e *nonRetryableError) Unwrap() error {
+	if e == nil {
+		return nil
+	}
+	return e.Err
+}
+
+func isNonRetryableError(err error) bool {
```

Comment:
Export and add godocs? We exported the constructor/type (depending on which suggestions you take).